DemoApp/Assets/Scripts/HomeScreen.cs

```diff
@@ -1,18 +1,234 @@
+
+        // Load the first RewardedAd
+        LoadRewardedAd();
+    }
+
+    private void LoadRewardedAd()
+    {
+        rewardedStatusText.text = "Loading...";
+        MaxSdk.LoadRewardedAd(RewardedAdUnitId);
+    }
+
+    private void ShowRewardedAd()
+    {
+        if (MaxSdk.IsRewardedAdReady(RewardedAdUnitId))
+        {
+            rewardedStatusText.text = "Showing";
+            MaxSdk.ShowRewardedAd(RewardedAdUnitId);
+        }
+        else
+        {
+            rewardedStatusText.text = "Ad not ready";
+        }
+    }
+
+    private void OnRewardedAdLoadedEvent(string adUnitId)
+    {
+        // Rewarded ad is ready to be shown. MaxSdk.IsRewardedAdReady(rewardedAdUnitId) will now return 'true'
+        rewardedStatusText.text = "Loaded";
+        Debug.Log("Rewarded ad loaded");
+    }
+
+    private void OnRewardedAdFailedEvent(string adUnitId, int errorCode)
+    {
+        // Rewarded ad failed to load. We recommend re-trying in 3 seconds.
+        rewardedStatusText.text = "Failed load: " + errorCode + "\nRetrying in 3s...";
+        Debug.Log("Rewarded ad failed to load with error code: " + errorCode);
+        Invoke("LoadRewardedAd", 3);
+    }
+
+    private void OnRewardedAdFailedToDisplayEvent(string adUnitId, int errorCode)
+    {
+        // Rewarded ad failed to display. We recommend loading the next ad
+        Debug.Log("Rewarded ad failed to display with error code: " + errorCode);
+        LoadRewardedAd();
+    }
+
+    private void OnRewardedAdDisplayedEvent(string adUnitId)
+    {
+        Debug.Log("Rewarded ad displayed");
+    }
+
+    private void OnRewardedAdClickedEvent(string adUnitId)
+    {
+        Debug.Log("Rewarded ad clicked");
+    }
+
+    private void OnRewardedAdDismissedEvent(string adUnitId)
+    {
+        // Rewarded ad is hidden. Pre-load the next ad
+        Debug.Log("Rewarded ad dismissed");
+        LoadRewardedAd();
+    }
+
+    private void OnRewardedAdReceivedRewardEvent(string adUnitId, MaxSdk.Reward reward)
+    {
+        // Rewarded ad was displayed and user should receive the reward
+        Debug.Log("Rewarded ad received reward");
+    }
+
+    #endregion
+
+    #region Banner Ad Methods
+
+    private void InitializeBannerAds()
+    {
+        // Banners are automatically sized to 320x50 on phones and 728x90 on tablets
+        // You may use the utility method `MaxSdkUtils.isTablet()` to help with view sizing adjustments
+        MaxSdk.CreateBanner(BannerAdUnitId, MaxSdkBase.BannerPosition.BottomCenter);
+
+        // Set background or background color for banners to be fully functional
+        MaxSdk.SetBannerBackgroundColor(BannerAdUnitId, Color.black);
+    }
+
+    private void ToggleBannerVisibility()
+    {
+        if (!isBannerShowing)
+        {
+            MaxSdk.ShowBanner(BannerAdUnitId);
+            isBannerShowing = true;
+            showBannerButton.GetComponentInChildren<Text>().text = "Hide Banner";

```

Comment:
I feel like there was a cleaner way to set button text but can't recall off the top of my head

Reply:
Everything I saw from Google was using this method or a similar one. If you know of a cleaner way I'll be happy to use it!

Reply:
yup fair